File: Pythonleetcode/Leetcode112.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def hasPathSum(self, root, sum):
        """
        :type root: TreeNode
        :type sum: int
        :rtype: bool
        """
        if root is None:
            return False
        elif  root.left is None and root.right is None and root.val==sum :
            return True
        else:
            return self.hasPathSum(root.right,sum-root.val) or self.hasPathSum(root.left,sum-root.val)

File: CppPractice/CppLeetcode/Leetcode112.cpp
 
class Solution {
public:
    int target; 
    bool helper(TreeNode *node,int temp)
    {
        if(!node->left && !node->right){return temp+node->val==target;}
        else if(!node->left){return helper(node->right,temp+node->val);}
        else if(!node->right){return helper(node->left,temp+node->val);}
        else{return helper(node->left,temp+node->val)||helper(node->right,temp+node->val);}

    }
    bool hasPathSum(TreeNode* root, int targetSum) 
    {
        if(!root){return false;}
        target = targetSum;
        return helper(root,0);
    }
};